components/datetime/src/provider/neo.rs

```diff
@@ -588,25 +604,24 @@ impl AsULE for SkeletonDataIndex {
         flags |= (self.has_long as u8) << 7;
         flags |= (self.has_medium as u8) << 6;
         flags |= (self.has_plurals as u8) << 5;
-        SkeletonDataIndexULE([flags, self.index])
+        SkeletonDataIndexULE(flags)
     }
 
     fn from_unaligned(unaligned: Self::ULE) -> Self {
-        let [flags, index] = unaligned.0;
+        let flags = unaligned.0;
         // TODO: `flags` could have more bits set, but we don't check that here.
         SkeletonDataIndex {
             has_long: (flags & (1 << 7)) != 0,
             has_medium: (flags & (1 << 6)) != 0,
             has_plurals: (flags & (1 << 5)) != 0,
-            index,
         }
     }
 }
 
 #[icu_provider::data_struct(
     marker(
         GregorianDateNeoSkeletonPatternsV1Marker,
-        "datetime/patterns/gregory/date_skeleton@1"
+        "datetime/patterns/gregory/skeleton@1"
```

Comment:
question (non blocking): I forget; is there a reason this is gregorian-only? Do we plan to add more or is this now shared between calendars?

Reply:
I plan to add `"datetime/patterns/buddhist/skeleton@1"`, ... using all of our existing calendar-specific-key infrastructure